### PR TITLE
SPM-1165: Added LRU cache for advisory detail endpoint

### PR DESCRIPTION
**Pull request by Josca**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: manager/controllers/advisory_detail.go [hunk 1]
```diff
@@ -53,28 +57,42 @@ func AdvisoryDetailHandler(c *gin.Context) {
...
+
+func getAdvisoryFromDB(advisoryName string) (*AdvisoryDetailResponse, error) {
 	var advisory models.AdvisoryMetadata
 	err := database.Db.Where("name = ?", advisoryName).First(&advisory).Error
+	if err != nil {
+		return nil, err
+	}
+
 	if errors.Is(err, gorm.ErrRecordNotFound) {
-		LogAndRespNotFound(c, err, "advisory not found")
```
Comment (MichaelMraka) [inline manager/controllers/advisory_detail.go:59]: This is a dead code. On L77 you return if `err != nil`  return so here `err` can be only `nil` and never `gorm.ErrRecordNotFound`.

File: manager/controllers/advisory_detail.go [hunk 2]
```diff
@@ -53,28 +57,42 @@ func AdvisoryDetailHandler(c *gin.Context) {
...
-		return
+		return nil, nil
 	}
 
 	if err != nil {
-		LogAndRespError(c, err, "database error")
```
Comment (MichaelMraka) [inline manager/controllers/advisory_detail.go:64]: Same here, if `err != nil` you already returned from the function on L77.

**Review by MichaelMraka — CHANGES_REQUESTED** [review]

File: manager/controllers/advisory_detail.go (hunk 1 above)
Comment (Josca) [reply manager/controllers/advisory_detail.go:59]: yes, thanks

File: manager/controllers/advisory_detail.go (hunk 2 above)
Comment (Josca) [reply manager/controllers/advisory_detail.go:64]: yes, thanks

**Review by MichaelMraka — APPROVED** [review]

**Pull request merged**